### ISSUE TITLE
[a11y][WCAG 2.4.3]: Linear navigation focus is not visible on checkbox controls

### ISSUE THREAD

**Issue by dkatzz** [civiform/civiform#9914]
Observed Results:
--
While navigating in Linear Navigation (Search + right/left) arrow keys, screen reader foucs is not visible on the checkbox controls rather it is displaying somewhere else in the page.

Screencast: Chromebook + Chromevox[here]( https://drive.google.com/file/d/1q6IQ53AYKNJU-R3cTKZ8VzSLOjwyoKVc/view?usp=drive_link ) \
Screencast: Windows + NVDA[here]( https://screencast.googleplex.com/cast/NjcyMjI2NzUyMTIyMDYwOHw3YmM0ZGYyMi05YQ )

User Impact:
-----
Users who are low vision and depend on screen readers will face difficulty or get confused if focus is not on the controls while navigating.

Expected Results:
--
While navigating using linear navigation (Search+right/left) arrow keys, checkbox controls under "Available programs and services" region,screen reader focus should visible on the checkbox control and name of the checkbox control individually

Steps to Reproduce:
-------
0. Enable Chromevox using Ctrl+Alt+Z
1. Launch the URL[here]( https://ns.civiform.dev/programs )
2. Using Linear Navigate(Search +left/right arrow) over the Checkbox controls at "Available programs and services - filter by category" region.
3. Observe the linear navigation focus on checkbox controls

WCAG Details:
--
[WCAG 2.4.3][Focus-Order]( https://www.w3.org/WAI/WCAG22/quickref/?versions=2.0#focus-order )

Additional Information
--
N/A

Environment Details:
--
Machine: chromebook 13.3/Windows \
Browser: Version 132.0.6834.208 (Official Build) (64-bit) \
Screenreader: Chromevox/NVDA 2024.4.2135031 \
Bug Filling Time: 15 mins.

**Comment by dkatzz** [comment]
Second screencast:

https://github.com/user-attachments/assets/09fe1cae-e884-4755-a846-5dad139b73ad

**Comment by melanie-exygy** [comment]
Tested on NVDA and had a similar experience, with a small difference depending on whether I was using the up/down arrows or form navigation (F and Shift-F).  I believe with NVDA, most users would be using form navigation.  With the arrow keys, the red focus box only appear after a second click.  With form navigation, the red focus box doesn't appear at all.

Note that I tested this on a checkbox question within the application form and had the same experience.

https://github.com/user-attachments/assets/e304969b-ceda-49ba-af0b-eb3d69497b9d

**Comment by melanie-exygy** [comment]
Also, tested with Narrator and Edge browser.  The experience was similar to Chromevox.
